packages/db/discordBot.ts

```diff
@@ -104,11 +110,85 @@ bot.on("threadUpdate", async (oldThread, newThread) => {
 });
 
 bot.on("interactionCreate", async (interaction) => {
-  if (!interaction.isChatInputCommand()) return;
-
-  if (interaction.commandName === "ping") {
+  if (interaction.isChatInputCommand() && interaction.commandName === "ping") {
     await interaction.reply("Pong!");
   }
+
+  if (interaction.isButton() && interaction.customId === "resolveThread") {
+    try {
+      // Update button to in-progress state.
+      await interaction.update({
+        components: [
+          {
+            type: ComponentType.ActionRow,
+            components: [
+              {
+                type: ComponentType.Button,
+                label: "Working on it...",
+                style: ButtonStyle.Secondary,
+                customId: "resolveThread",
+                disabled: true,
+              },
+            ],
+          },
+        ],
+      });
```

Comment:
can we just have it resolve the thread here? Then the update might automatically propagate to the bridge. I originally thought the button would come in via a webhook straight to the convex backend (url button) but maybe just doing it all in the bot is easier / faster